src/game.cpp

```diff
@@ -851,6 +852,77 @@ bool game::start_game()
     return true;
 }
 
+static bool compare_distance( const tripoint &goal_a, const tripoint &goal_b )
+{
+    tripoint origin = g->u.global_omt_location();
+    return rl_dist( origin, goal_a ) < rl_dist( origin, goal_b );
+}
+
+void game::place_starting_vehicle( cata::optional<vproto_id> starting_vehicle )
+{
+    if( !starting_vehicle ) {
+        return;
+    }
+    vehicle veh( *starting_vehicle );
+    std::vector<std::string> omt_search_types;
+    if( veh.can_float() ) {
+        omt_search_types.push_back( "river" );
+        omt_search_types.push_back( "lake" );
+    }
+    if( veh.max_ground_velocity() > 0 ) {
+        omt_search_types.push_back( "road" );
+        omt_search_types.push_back( "field" );
+    }
+    if( omt_search_types.empty() ) {
+        return;
+    }
+    find_location_for_starting_vehicle( *starting_vehicle, omt_search_types );
+
+}
+
+void game::find_location_for_starting_vehicle( vproto_id id,
+        std::vector<std::string> omt_search_types )
+{
+    for( const std::string search_type : omt_search_types ) {
+        omt_find_params find_params;
+        find_params.must_see = false;
+        find_params.cant_see = false;
+        find_params.types.emplace_back( search_type, ot_match_type::type );
+        for( int count = 0; count < 10; ++count ) {
+            // find nearest road
+            find_params.min_distance = count * 2;
+            find_params.search_range = 10 + count * 2;
+            // if player spawns underground, park their car on the surface.
+            tripoint omt_origin = u.global_omt_location();
+            omt_origin.z = 0;
+            std::vector<tripoint> goals = overmap_buffer.find_all( omt_origin, find_params );
+            std::sort( goals.begin(), goals.end(), compare_distance );
+            for( const tripoint &goal : goals ) {
+                if( goal != overmap::invalid_tripoint ) {
+                    // try place vehicle there.
+                    tinymap target_map;
+                    target_map.load( omt_to_sm_copy( goal ), false );
+                    tripoint origin = target_map.getlocal( sm_to_ms_copy( omt_to_sm_copy( goal ) ) ) + point(
+                                          SEEX, SEEY );
+                    vehicle *veh = target_map.add_vehicle( id, origin, rng( -90, 90 ), rng( 50, 80 ),
```

Comment:
I guess axis-aligned vehicles would be better: looks better, and zeds can't get inside.